platforms/src/Twitter/procedures/twitterOauth.ts

```diff
@@ -68,54 +88,125 @@ export const getAuthClient = async (sessionKey: string, code: string, context: T
+    context.twitter.id === undefined ||
+    context.twitter.username === undefined
+  ) {
+    try {
+      // return information about the (authenticated) requesting user
+      const user = await twitterClient.users.findMyUser({
+        "user.fields": ["created_at"],
+      });
+
+      if (!context.twitter) context.twitter = {};
+
+      context.twitter.createdAt = user.data.created_at;
+      context.twitter.id = user.data.id;
+      context.twitter.username = user.data.username;
+      return {
+        createdAt: context.twitter.createdAt,
+        id: context.twitter.id,
+      };
+    } catch (_error) {
+      const error = _error as ProviderError;
+      if (error?.response?.status === 429) {
+        return {
+          errors: ["Error getting getting Twitter info", "Rate limit exceeded"],
+        };
+      }
+      return {
+        errors: ["Error getting getting Twitter info", `${error?.message}`],
+      };
+    }
+  }
+  return {
+    createdAt: context.twitter.createdAt,
+    id: context.twitter.id,
+    username: context.twitter.username,
+  };
 };
 
-export const getFollowerCount = async (twitterClient: Client): Promise<TwitterFollowerResponse> => {
-  // public metrics returns more data on user
-  const myUser = await twitterClient.users.findMyUser({
-    "user.fields": ["public_metrics"],
-  });
+export const getUserTweetTimeline = async (
+  context: TwitterContext,
+  twitterClient: Client
+): Promise<UserTweetTimeline> => {
+  let nextToken: string | undefined;
+  let tweetDays: Set<string>;
+  if (context.twitter.numberDaysTweeted === undefined || context.twitter.valid === false) {
+    try {
+      // return information about the (authenticated) requesting user
+      const userId = (await twitterClient.users.findMyUser()).data.id;
+
+      if (!context.twitter) context.twitter = {};
+      // returns user tweet information
+      do {
+        const userTweetDaysResponse = await twitterClient.tweets.usersIdTweets(userId, {
+          max_results: 100,
+          pagination_token: nextToken,
+          "tweet.fields": ["created_at"],
+        });
+        for (const tweet of userTweetDaysResponse.data) {
+          // Extract date from created_at
+          const date = new Date(tweet.created_at).toISOString().split("T")[0];
+          tweetDays.add(date);
+        }
+        nextToken = userTweetDaysResponse.meta.next_token;
+        // If user has already tweeted for 120 distinct days, no need to continue
+        if (tweetDays.size >= 120) {
+          context.twitter.id = userId;
+          context.twitter.numberDaysTweeted = tweetDays.size;
+          context.twitter.valid = true;
+          return {
+            id: userId,
+            numberDaysTweeted: tweetDays.size,
+            valid: true,
+          };
+        }
+        // Respect rate limits by sleeping for a short time after each request
+        await new Promise((resolve) => setTimeout(resolve, 1000));
+      } while (nextToken);
```

Comment:
Seems like an edge case since the user would have to tweet a ton I think 11ish times a day(1400/120=11.6)?  but maybe it would be worth while limiting this to 14 calls to avoid rate limiting?

Reply:
Yeah, that makes total sense, I'll switch this up!